src/streamlink/plugins/trovo.py

```diff
@@ -0,0 +1,167 @@
+                                    "streamerInfo": {
+                                        "userName": str,
+                                    },
+                                    "vodInfo": {
+                                        "playInfos": [{
+                                            "desc": validate.all(validate.transform(lambda s: s.lower()), str),
+                                            "playUrl": validate.url(),
+                                        }],
+                                        "vid": str,
+                                        "title": str,
+                                        "categoryName": str,
+                                        "playbackRights": {
+                                            "playbackRightsSetting": str,
+                                            "playbackRights": str,
+                                        },
+                                    },
+                                },
+                            }, {}),
+                        },
+                    },
+                }],
+                validate.get((0, "data", "batchGetVodDetailInfo", "VodDetailInfos"))
+            ),
+        )
+
+        if video_id not in json:
+            log.error("Video not found")
+            return
+        else:
+            json = json[video_id]
+
+        log.debug(json["vodInfo"]["playbackRights"])
+        self.id = json["vodInfo"]["vid"]
+        self.author = json["streamerInfo"]["userName"]
+        self.title = json["vodInfo"]["title"]
+        self.category = json["vodInfo"]["categoryName"]
+
+        for s in json["vodInfo"]["playInfos"]:
+            q = s["desc"]
+            if "(source)" in q:
+                q = f"source_{q.replace('(source)', '')}"
+            yield q, HLSStream(self.session, s["playUrl"])
+
+    def get_live(self, user):
+        json = self.session.http.post(
+            f"https://api-web.trovo.live/graphql?qid={self.generate_qid()}",
+            json=[{
+                "operationName": "live_LiveReaderService_GetLiveInfo",
+                "variables": {
+                    "params": {
+                        "userName": user,
+                    },
+                },
+            }],
+            schema=validate.Schema(
+                validate.parse_json(), validate.any([{
+                    "data": {
+                        validate.optional("live_LiveReaderService_GetLiveInfo"): {
+                            "streamerInfo": {
+                                "userName": str,
+                            },
+                            "programInfo": {
+                                "id": str,
+                                "title": str,
+                                "streamInfo": [{
+                                    "desc": validate.all(validate.transform(lambda s: s.lower()), str),
+                                    "playUrl": validate.transform(lambda s: s.replace(".flv?", ".m3u8?")),
```

Comment:
The site is using a progressive HTTP stream with a Flash video container. Is this only the case for non-authenticated users? I'm asking because the site asks for a login to be able to watch source quality streams. And VODs are available in HLS regardless.

Reply:
I don't know about logged-in verses not, as I have not created an account.  I thought it would be better to use HLS rather than an HTTP stream, but given the A/V de-sync issue you have noticed (I had not, although I did not watch the output for very long), perhaps it would be better to just HTTP stream the FLV content?

Reply:
A progressive HTTP stream would definitely be worse. I've checked two streams, and both had desync with their HLS streams. It could be possible though that it was also the case on the site, which I didn't check. I will take another look later. If there is indeed a desync issue, then ffmpeg could help.

Reply:
How much de-sync were you seeing?  I've watched a couple of streams for a bit and cannot seem to see an obvious de-sync.  I was watching streams with player cams shown as a picture in picture.  It's not always easy to tell, but generally the A/V looked to be in sync as far as I could see.

Reply:
It could've been the streams themselves which I checked yesterday. There definitely was a delay of 2 seconds or so. I can't see anything like this right now, so I really don't know what was going on there.

Apart from the massive delay caused by the ridiculous targetduration on the playlists, the HLS streams should be working fine, I guess.